mscore/scoreview.cpp

```diff
@@ -3853,7 +3853,13 @@ void ScoreView::adjustCanvasPosition(const Element* el, bool playBack, int staff
       if (oldX == x && oldY == y)
             return;
 
-      setOffset(-x * physicalZoomLevel(), -y * physicalZoomLevel());
+      int cx = 0, cy = 0;
+      constraintCanvas(&cx, &cy);
+      x *= -physicalZoomLevel();
+      y *= -physicalZoomLevel();
+      cx = (x < 0) ? x : cx + _matrix.dx();
```

Comment:
Any special reason not to calculate and use the y offset too? Like for horizontal scrolling?

Reply:
Y'know, I didn't really think about the y-constraint since I was mainly concerned with the limiting of the horizontal borders in page mode and vertical scrolling. Not sure if there'd be any negative repercussions by trying the equivalent (but guessing *probably* not):

```cpp
cy = (y < 0) ? y : cy + _matrix.dy();
setOffset(cx, cy);
```

I'll check it out later

Reply:
Dunno ;-)

Reply:
Fwiw, I don't recall experiencing an issue with y-constraints, but I usually use vertical scrolling and limit scroll area options, so not sure if there's anything to be fixed... ;)

Reply:
Let's put it this way: in master there's `emit offsetChanged(_matrix.dx(), _matrix.dy());`, in 2 places, so both get updated?